src/libsyntax/parse/parser.rs

```diff
@@ -572,9 +572,13 @@ impl<'a> Parser<'a> {
                 self.bug("ident interpolation not converted to real token");
             }
             _ => {
-                let token_str = self.this_token_to_string();
-                Err(self.fatal(&format!("expected ident, found `{}`",
-                                    token_str)))
+                let mut err = self.fatal(&format!("expected identifier, found `{}`",
+                                                  self.this_token_to_string()));
+                if self.token == token::Underscore {
+                    err.fileline_note(self.span, "`_` is not an identifier but a wildcard pattern \
```

Comment:
I don't know, the compiler uses "wildcard", [wikipedia](https://en.wikipedia.org/wiki/Pattern_matching) uses it too and that part of the reference looks ancient and not well-maintained.